trunk/source/CDTParser/src/de/uni_freiburg/informatik/ultimate/cdt/parser/CDTParser.java

```diff
@@ -183,6 +191,29 @@ public IElement parseAST(final File[] files) throws Exception {
 		return new WrapperNode(null, decorator);
 	}
 
+	/**
+	 * Dumps the AST of a specified translation unit and report AST problem nodes.
+	 * 
+	 * @param tu Translation unit of a parsed input file.
+	 * @param printFunc Operation that prints each {@link IASTNode} to a {@link PrintStream}.
+	 * @param desc Description of printing operation.
+	 */
+	public void dumpAST(final IASTTranslationUnit tu, final BiConsumer<IASTNode, PrintStream> printFunc, final String desc) {
+
+		if (mLogger.isDebugEnabled()) {
+			final String fileName = normalizeCdtFilename(tu.getFilePath());
+			final OutputStream printOutput = new ByteArrayOutputStream();
+			final PrintStream printStream = new PrintStream(printOutput);

```

Comment:
**optional:** You could also use the `de.uni_freiburg.informatik.ultimate.core.lib.util.LoggerOutputStream` class.

Reply:
Thanks for the hint. If possible, I will replace the boiler plate code for the output stream creation with the `LoggerOutputStream` of the Ultimate core utils.